### ISSUE TITLE
What if #labels in training set is different than #labels in test set?

### ISSUE THREAD

**Issue by hdm30** [Extreme-classification/ECLARE#6]
Hi,

I realized that the datasets used for XMC models have the same number of labels in both training & test set. I'm wondering what happens if they are different. I tried to run ECLARE model on a custom dataset that satisfies the previous condition (#labels_train != #labels_test) and the mismatch dimension issue crops up. 

Why should #labels_train & #labels_test be equal? Do I need to tweak the code to resolve this problem? 
Many thanks,

**Comment by kunaldahiya** [comment]
Hi

ECLARE is primarily designed for supervised learning scenarios, i.e., every label should have at least one training point. So, a label that appears only in the test set (zero-shot labels) would be required to handle separately. The opposite scenario, i.e., if a label appears just in train set is trivial to handle.
